### ISSUE TITLE
compile errors on nistclock

### ISSUE THREAD

**Issue by rlust** [mrkale/NodeMCU-Modules#1]
I get several compile error when loading nistclock.

nistclock.lua:156: unexpected symbol near 'end'
stack traceback:
    [C]: in function 'dofile'
    stdin:1: in main chunk

**Comment by mrkale** [comment]
Do you run the module (along side with applicatioin scripts) in an ESP with sufficient free space on the file system? I have encountered a problem with ESP8266-01, where an uploaded file had been truncated without warning. Check module's file size in your ESP in comparison to the size in your workstation. Please read some comments about that issue in my [other repository](https://github.com/mrkale/NodeMCU-WifiDoubleSwitch/tree/ESP8266-01).

**Comment by rlust** [comment]
Yes I am running this along the application scripts. I have an ESP12E so I have plenty of space.
I am opening nistclock.lua along with the other files and sending all to ESP. It fails with about 5 errors when loading to ESP with ESPLORER. 
I am trying to get your other application WifiDouble Switch
nistclock.lua:165: unexpected symbol near 'end'
stack traceback:
    [C]: in function 'dofile'
    stdin:1: in main chunk

**Comment by mrkale** [comment]
I use ESPlorer as well. I used to get such errors at uploading to ESP8266-01, but not at  ESP8266-12E (Lolin NodeMCU). The errors were caused by script truncating.

The uploading should not fail with any error. Did you get this after uploading nistclock.lua?

``` lua
> Uploading to ESP file nistclock.lua...Success
```

Did you get this after compilation?

``` lua
> node.compile("nistclock.lua")
...
nistclock.lc    : 4744 bytes
nistclock.lua   : 7578 bytes
```

**Comment by rlust** [comment]
I cannot get a clean upload, nor a compile.
How do I fix script truncating? 
I have an AMICA 12-E, thanks for your help! I am so close and want to get this running.

**Comment by mrkale** [comment]
Try reflashing your devkit. I used to have similar problems with custom builds from the dev branch at begining the serie 1.x+. Download your [custom build](http://nodemcu-build.com/) and flash an integer version. The default configuration is good enough.

**Comment by rlust** [comment]
Can you check your code in nistclock.lua? This is where I am getting error:
I changed the code to get it to run, please let me know if what I fixed is wrong and now causing another error.
Thanks,

return
    tt[6],
    tt[5],
    tt[4],
    dayOfWeek(tt[1], tt[2], tt[3]),
    tt[3],
    tt[2],
    tt[1],
    tt[7]
end
[nistclock.lua.zip](https://github.com/mrkale/NodeMCU-Modules/files/129310/nistclock.lua.zip)

**Comment by mrkale** [comment]
Your code change is incorrect. Now the function _nistclock.getTime_ does not return _day, month, year, and julian day anymore_. The day, month, year are then needed in _init.lua_ for formatting the datetime string, but the _nistclock.getTime_ returns nil instead of them. Nil values in the string.format function cause the panic error.

You'd better to fork my repository and make experiments on your forked repo. If you had some functional code change or improvement, you might send me a pull request.
